.github/workflows/article-publisher.yml

```diff
@@ -0,0 +1,37 @@
+name: Articles Publisher
+
+on:
+  pull_request:
+    types: [closed]
+
+env:
+  HASHNODE_API_TOKEN: ${{ secrets.HASHNODE_API_TOKEN }}
+  HASHNODE_PUBLICATION_ID: ${{ secrets.HASHNODE_PUBLICATION_ID }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  BRANCH_NAME: "develop"
+
+jobs:
+  article-publisher:
+    if: github.event.pull_request.merged == true && github.event.pull_request.base.ref == 'develop'
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+
+      - name: install deps and build
+        run: |
+          cd ./publisher
+          npm install
+          npm run build
```

Comment:
Build should have happened in the previous build. I think we can just wait for it here.

Reply:
Hmm I don't quite get which build do you mean, if you mean _build-and-deploy_ in the _CI_ workflow here's why I didn't go with that approach : 

I actually don't want this job to depend on _build-and-deploy_ because _build-and-deploy_ runs on every push to any branch. 

Blog publishing will only work for merged PRs, in case you want to do editing or other peers want to double check the blog before publishing it.

So in this workflow we only download the dependencies necessary for publishing blogs and run the script to do it, we don't really need to wait on _build-and-deploy_.